### PR TITLE
[Enhancement] auto generate pk dump when tablet in error state

### PR DESCRIPTION
**Pull request by luohaha**
Why I'm doing:
In previous pr #38297 , we support dump pk tablet's all information for debug purpose. We can trigger pk dump by script cmd. And now I want to support auto pk dump when tablet is in error state.

What I'm doing:
1. Support auto generate pk dump when tablet in error state, add BE config `pk_dump_interval_seconds` to avoid generate too much dump files.
2. Change pk dump file path to `storage/tmp`.

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/storage/tablet_updates.cpp [hunk 1]
```diff
@@ -2285,6 +2285,26 @@ bool TabletUpdates::check_rowset_id(const RowsetId& rowset_id) const {
...
+                          << " generate pk dump success, path : " << pkd.dump_filepath();
+            }
+        }
+    }
+    return Status::OK();
+}
+
 void TabletUpdates::remove_expired_versions(int64_t expire_time) {
     if (_error) {
         LOG(WARNING) << strings::Substitute(
```
[the remark this answers is not in the document]
Comment (luohaha) [reply be/src/storage/tablet_updates.cpp:2310]: Only in one thread.

**Review by decster — DISMISSED** [review]

**Review by wyb — APPROVED** [review]

**Review by decster — APPROVED** [review]

File: be/src/storage/tablet_manager.cpp [hunk 2]
```diff
@@ -701,6 +701,27 @@ TabletSharedPtr TabletManager::find_best_tablet_to_compaction(CompactionType com
     return best_tablet;
 }
 
+Status TabletManager::generate_pk_dump_in_error_state() {
```
Comment (kevincai) [inline be/src/storage/tablet_manager.cpp:704]: it is a little confusing reading from this function name, it says `generate_pk_dump_in_error_state`, but actually, it pulls all pk tablets without filtering their status.

And the status is filtered in `generate_pk_dump` which is actually only do the dump if tablet is error, but its name doesn't say so.

**Review by kevincai — APPROVED** [review]
tiny comment.

File: be/src/storage/tablet_manager.cpp (hunk 2 above)
Comment (luohaha) [reply be/src/storage/tablet_manager.cpp:704]: Ok, i will change this name in next pr.

**Pull request merged**